### PR TITLE
fix: Toolbar better auth issue

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

When the toolbar gets logged out its super unclear.

## Changes

* Adds an authenticate button so its really obvious
* Changes the logic to detect 401 always and trigger the logout
* Removes some old code

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

* Manually broke the temporary token locally so that we get the 401 message and validated it worked as expected.

### CODE REVIEW

File: frontend/src/toolbar/bar/Toolbar.tsx [hunk 1]
```diff
@@ -178,19 +184,32 @@ export function Toolbar(): JSX.Element {
...
+                </ToolbarButton>
                 {isAuthenticated ? (
                     <>
-                        <ToolbarButton icon={<IconSearch />} menuId="inspect" />
-                        <ToolbarButton icon={<IconCursorClick />} menuId="heatmap" />
-                        <ToolbarButton icon={<IconBolt />} menuId="actions" />
-                        <ToolbarButton icon={<IconToggle />} menuId="flags" title="Feature flags" />
+                        <ToolbarButton menuId="inspect">
+                            <IconSearch />
+                        </ToolbarButton>
```
Comment (pauldambra) [inline frontend/src/toolbar/bar/Toolbar.tsx:197]: is the icon change some react foo i don't understand or just its pleasing to your eye (or something else?) :)

File: frontend/src/toolbar/toolbarConfigLogic.ts [hunk 2]
```diff
@@ -120,9 +120,12 @@ export async function toolbarFetch(
     if (response.status === 403) {
         const responseData = await response.json()
         // Do not try to authenticate if the user has no project access altogether
```
Comment (pauldambra) [inline frontend/src/toolbar/toolbarConfigLogic.ts:122]:
```suggestion
```

no need for a comment here IMHO

**Review by pauldambra — APPROVED** [review]
:shipit:

File: frontend/src/toolbar/bar/Toolbar.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/toolbar/bar/Toolbar.tsx:197]: It's just there is no difference between the text or an icon - they were just children so I made it standard children

**Pull request merged**